src/js/components-react/colorPickerWrap.js

```diff
@@ -0,0 +1,120 @@
+import React, {PureComponent} from 'react';
+import debounce from 'lodash/debounce';
+import color from 'react-color/lib/helpers/color';
+
+export const ColorWrap = (Picker) => {
+  const COLOR_EMPTY = {
+    hex: '',
+    hsl: {a: '',
+      h: '',
+      l: '',
+      s: '',
+    },
+    hsv: {a: '',
+      h: '',
+      s: '',
+      v: '',
+    },
+    oldHue: '',
+    rgb: {a: '',
+      b: '',
+      g: '',
+      r: '',
+    }
+  };
+  class ColorPicker extends PureComponent {
+
+    constructor(props) {
+      super();
+      if (props.hasOwnProperty('color')) {
+        this.state = {
+          ...color.toState(props.color, 0),
+        };
+      } else {
+        this.state = COLOR_EMPTY;
+      }
+
+      this.debounce = debounce((fn, data, event) => {
+        fn(data, event);
+      }, 100);
+    }
+
+    // eslint-disable-next-line react/no-deprecated
+    componentWillReceiveProps(nextProps) {
```

Comment:
Please, do not use componentWillReceiveProps.
It is deprecated and soon will be entirely removed from React, and is considered an anti-pattern.
Consider refactoring to functional component and write sample for React as a functional component.
Thank you!

Reply:
>Please, do not use componentWillReceiveProps.
I have removed componentWillReceiveProps function.

>Consider refactoring to functional component
It is necessary to use state inside colorpicker, such as 'status' to judge the window open or not.
So i think it's difficult to refactor to a stateless functional component.
Please confirm it and merge the source.
Thank you!